src/pages/workspace/card/issueNew/LimitTypeStep.tsx

```diff
@@ -1,28 +1,75 @@
-import React from 'react';
+import React, {useCallback, useMemo, useState} from 'react';
 import {View} from 'react-native';
-import FormProvider from '@components/Form/FormProvider';
+import type {OnyxEntry} from 'react-native-onyx';
+import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import InteractiveStepSubHeader from '@components/InteractiveStepSubHeader';
 import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
 import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as Card from '@userActions/Card';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type * as OnyxTypes from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
-function LimitTypeStep() {
+type LimitTypeStepProps = {
+    // The policy that the card will be issued under
+    policy: OnyxEntry<OnyxTypes.Policy>;
+};
+
+function LimitTypeStep({policy}: LimitTypeStepProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
+    const [issueNewCard] = useOnyx(ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD);
+    const [typeSelected, setTypeSelected] = useState(issueNewCard?.data?.limitType);
 
-    const submit = () => {
-        // TODO: the logic will be created in https://github.com/Expensify/App/issues/44309
-        Card.setIssueNewCardStep(CONST.EXPENSIFY_CARD.STEP.LIMIT);
-    };
+    const submit = useCallback(() => {
+        Card.setIssueNewCardDataAndGoToStep({limitType: typeSelected}, CONST.EXPENSIFY_CARD.STEP.LIMIT);
+    }, [typeSelected]);
 
-    const handleBackButtonPress = () => {
+    const handleBackButtonPress = useCallback(() => {
         Card.setIssueNewCardStep(CONST.EXPENSIFY_CARD.STEP.CARD_TYPE);
-    };
+    }, []);
+
+    const areApprovalsConfigured = !isEmptyObject(policy?.approver) && policy?.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL;
+
+    const data = useMemo(() => {
+        const options = [];
+
+        if (areApprovalsConfigured) {
```

Comment:
Come from the design docs:

> We will display and check the Smart Limit option as the default selection if the company has the approvals configured

As I understand, we always display three options and if areApprovalsConfigured is true the LIMIT_TYPES.SMART will be **selected** by default

Reply:
yep, but under that:

>When approvals are not configured, we will not show `Smart Limit` as an option, and default to the`Monthly` limit type. Each card assigned to a single user will have an independent Smart Limit. 

<img width="720" alt="Screenshot 2024-07-04 at 12 45 08" src="https://github.com/Expensify/App/assets/28020445/7328cbde-33d8-4f1f-a7db-440c67d4b141">

but yeah, I will make the smart option and monthly selected depending on the approvals configuration